modules/renter/proto/merkleroots.go

```diff
@@ -66,27 +81,26 @@ func loadExistingMerkleRoots(file *fileSection) (*merkleRoots, error) {
 	if err != nil {
 		return nil, err
 	}
-	// Get the size of the file and read it.
-	var size int64
-	if size, err = file.Size(); err != nil {
-		return nil, err
-	}
-	fileData := make([]byte, size)
-	if _, err := file.ReadAt(fileData, 0); err != nil {
-		return nil, err
-	}
 	// Read the roots from the file without reading all of them at once.
-	r := bytes.NewBuffer(fileData)
-	for i := 0; i < mr.numMerkleRoots; i++ {
-		var root crypto.Hash
-		if _, err = io.ReadFull(r, root[:]); err == io.EOF {
+	readOff := int64(0)
+	rootsData := make([]byte, rootsDiskLoadBulkSize)
+	for {
+		n, err := file.ReadAt(rootsData, readOff)
```

Comment:
are you not able to use `bufio` anymore because `file` only supports `ReadAt`, not `Read`?
That makes me sad.